docs: configure access for entra id users

flyinghermit opened the pull request.

flyinghermit committed:
grant teleport roles to entra id imported groups

flyinghermit committed:
updated copy

r0mant reviewed (commented):
Some suggestions to address, main points are:

- Remove app labels wildcards, show how to grant and use traits right away.
- Make sure to show trait templating via `{{external.xxx}}` syntax.

flyinghermit committed:
review comments:
- add screenshots
- start with traits template instead of late intro
- use tempalted role
- remove connector role mapper tip

flyinghermit commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,196 @@
+
+Repeat the steps and create another user group named `ad-app-admin`.
+
+In the next import cycle, Teleport will create an Access List for each of these groups.
```
Clarified its 5 mins.

flyinghermit commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
````diff
@@ -0,0 +1,196 @@
+$ tctl create support-team.yaml
+```
+
+<Admonition type="tip" title="Role resource label">
+  In the example role `app-monitor`, the allow `app_labels` 
+  rule we defined applies to the application resources. You may need to reference 
+  a different [resource label](../../reference/access-controls/roles.mdx#example-role-specification) 
+  rule if you are following through this guide with a different kind of resource.  
+</Admonition>
````
I think its still useful to reference other resource label?

flyinghermit commented:
Ok, after having written all the short term access guide, I just now realized that the Entra ID integration maps to requester role by default. With requester role, users can request access to resource so the whole idea of role access request makes a little sense!

flyinghermit commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
````diff
@@ -0,0 +1,177 @@
+```
+
+We will have two user groups created in Entra ID: `ad-app-admin` and 
+`ad-app-support`. We want members of these groups to have long-term and 
+short-term access to Grafana respectively.
````
Gotcha, will do that. I think I saw the "long-term" and "short-term" terminology in the works for the new access list getting started UI so thought to bring that here too.

flyinghermit commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,177 @@
+
+Here's what defined in the role spec-  we name this role as "app-monitor". 
+The role is configured with allowed `app_labels` that matches 
+label key `env` and a label value `'{{external.apps}}'`, which will be derived from 
+user's `apps` trait. As long as the label value configured in the application resource 
+label, and the value configured in the user `apps` trait matches, this role will grant 
+access to that application. 
```
Good point. I've added a small description to explain that. Feel free to suggest alternative.

> Using trait template to define label makes this role scalable as user traits are dynamically
> configurable.
> 
> When user authenticates with Entra ID, the SAML attributes (or claims if you are using OIDC) 
> that are available in the user SSO response are preserved as user traits. Additionally, you 
> can also grant traits to user using Access List. As you will see in the next step below, 
> this guide will use Access List to grant trait to the user.

flyinghermit committed:
review suggestions

flyinghermit commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,177 @@
+ - rbac
+---
+
+This guide shows how to configure access for Entra ID imported users with Nested Access Lists. 
```
Keeping this as it is for now as the nested access list is going to be the recommended way when using Entra ID integration. 

I initially had a commit with the similar copy but removed based on suggestion.
>  It's possible to configure Entra ID groups to Teleport role mapping 
>   from the Auth Connector settings. While it is easier to get started, 
>   we recommend Nested Access List setup because it offers greater flexibility 
>   over role and trait grants and can be periodically reviewed. 
> 
>   Entra ID also has a limit on [number of groups](https://learn.microsoft.com/en-us/entra/identity/hybrid/connect/how-to-connect-fed-group-claims#important-caveats-for-this-functionality)
>   that can be included in a SAML or an OIDC response. 
>   This can be prohibitive in a large-scale deployment.

flyinghermit committed:
Merge branch 'master' into sshah/entra-id-rbac

flyinghermit committed:
update role template guide link

r0mant commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,168 @@
+We will have two user groups created in Entra ID: `ad-app-admin` and 
+`ad-app-support`. We want members of these groups to have a permanent 
+access and an ability to request access to Grafana respectively. You 
+may use an existing Entra ID groups.
```
```suggestion
may also use existing Entra ID groups instead.
```

r0mant commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,168 @@
+access to that application. 
+
+Using trait template to define label makes this role scalable as user traits are dynamically
+configurable.
```
```suggestion
configurable. You can learn more about traits and role templates [in this guide](https://goteleport.com/docs/reference/access-controls/roles/#role-templates).
```

r0mant commented on docs/pages/identity-governance/entra-id/configure-access.mdx:
```diff
@@ -0,0 +1,168 @@
+![Teleport long term access list](../../../img/igs/entraid/entra-teleport-acl-long-term.png)
+
+This Access List grants `app-monitor` role and a trait `apps: monitor` to its members.
+Effectively, allowing its members to access resource application resource by the `app-monitor` role.
```
```suggestion
Effectively, allowing its members to access application resource via the `app-monitor` role.
```

flyinghermit committed:
apply review suggestions

flyinghermit committed:
Merge branch 'master' into sshah/entra-id-rbac
